err_stash.py

```diff
@@ -42,22 +47,125 @@ def fetch_repo_commits(self, project, slug, until, since):
+        except GithubException as e:
+            if e.status == 404:  # branch doesn't exist on this repo
+                # REMINDER: trying to match branch name like this:
+                # return [branch for branch in list(repo.get_branches()) if branch_name in branch.name]
+                # slows down the merge plans creation in ~5 to ~10 extra seconds.
+                return []
+            else:
+                raise
+
+    def delete_branch(self, organization:str, repo_name:str, branch_name:str):
+        """
+        Deletes one branch from one repository.
+        """
+        repo = self.repos[organization][repo_name]
+        try:
+            git_ref = repo.get_git_ref("heads/{branch_name}".format(branch_name=branch_name))
+        except GithubException as e:
+            raise CheckError(
+                "Error deleting branch '{branch_name}' in repo {repo_name}".format(
+                    branch_name=branch_name, repo_name=repo_name)
+            ) from e
+        else:
+            git_ref.delete()
+
+    def fetch_pull_requests(self, organization, repo_name):
+        """
+        Returns a list with all open pull requests
+        """
+        return list(self.repos[organization][repo_name].get_pulls())
+
+    def fetch_pull_request(self, organization, repo_name, pr_id):
+        """
+        Returns a specific github.PullRequest.PullRequest
+
+        :param str organization:
+            Name of the Github organization
+
+        :param str repo_name:
+            Name of the Github repository
+
+        :param int pr_id:
+            Pull request ID (the same you see on the Github PR page).
+        """
+        return self.repos[organization][repo_name].get_pull(pr_id)
+
+    def fetch_repo_commits(self, organization, repo_name, from_branch:str, to_branch:str):
+        """
+        Returns a list of commits that are on from_branch, but not in to_branch, i.e.,
+        commits that are added by the PR.
+
+        :param from_branch:
+             Name of the branch that created the PR (Github calls this head branch).
+
+        :param to_branch:
+            Name of the target branch in the PR (Github calls this base branch).
+        """
+        repo = self.repos[organization][repo_name]
+        return repo.compare(to_branch, from_branch).commits
+
+
 class MergePlan:
     """
     Contains information about branch and PRs that will be involved in a merge operation.
     """
 
-    def __init__(self, project, slug):
+    def __init__(self, project, slug, *, comes_from_github=False):
         self.project = project
         self.slug = slug
         self.branches = []
         self.pull_requests = []
         self.to_branch = None
+        self.comes_from_github = comes_from_github
 
 
 def get_self_url(d):
     """Returns the URL of a Stash resource"""
-    return d['links']['self'][0]['href']
+    return d.html_url if isinstance(d, PullRequest) else d['links']['self'][0]['href']
```

Comment:
But `d` is actually a `Stash` pull request, it's only doing this `isinstance` check because `Stash` sometimes uses paginated (dicts) objects and others their own `PullRequest` object.

Reply:
As we talked, this will be done on a future issue.